lib/memory_vty.c

```diff
@@ -25,7 +25,9 @@
 #include <malloc.h>
 #endif /* HAVE_MALLINFO */
 #include <dlfcn.h>
+#ifdef HAVE_DLINFO_LINKMAP
```

Comment:
I am not sure if it is the right fix here. TBC.

Reply:
I think it gets us closer to the right thing to do.